tests/unit/test_util.py

```diff
@@ -37,32 +37,46 @@ class FakeModel(FakeApiModelExtension):
     name: str
 
 
+class FakeModel2(FakeApiModelExtension):
+    code: int = 0
+    name: str
+    status: str
+
+
 class TestParseInput:
 
     def test_no_data(self, app):
         """
         GIVEN a request with no parameters of any type
         WHEN we call parse_input on that request
-        THEN we get a BaseApiModel returned with a code of 400
+        THEN we get a BaseApiView returned with a code of 400
         AND an error explaining what is wrong
         """
         with app.test_request_context():
             retval = parse_input(app.logger, FakeModel)
-            assert 'Field required' in retval.error
-            assert isinstance(retval, BaseApiModel)
+            assert (len(retval.errors)) == 1
+            assert 'name' == [*retval.errors][0]
+            assert 'Field required' == retval.errors.get('name')
+            assert isinstance(retval, BaseApiView)
 
-    def test_no_data_custom_return(self, app):
+    @pytest.mark.parametrize("to_parse,attrs_to_failed",
```

Comment:
suggestion: I think the naming of the param 'attrs_to_failed' is not very clear. Maybe just 'required_attrs'?

Reply:
Makes quite sense. Thanks!